src/CPPScope.cxx

```diff
@@ -531,13 +528,11 @@ static int meta_setattro(PyObject* pyclass, PyObject* pyname, PyObject* pyval)
 // the C++ side, b/c there is no descriptor yet. This triggers the creation for
 // for such data as necessary. The many checks to narrow down the specific case
 // are needed to prevent unnecessary lookups and recursion.
-    if (((CPPScope*)pyclass)->fFlags & CPPScope::kIsNamespace) {
     // skip if the given pyval is a descriptor already, or an unassignable class
-        if (!CPyCppyy::CPPDataMember_Check(pyval) && !CPyCppyy::CPPScope_Check(pyval)) {
-            std::string name = CPyCppyy_PyText_AsString(pyname);
-            if (Cppyy::GetNamed(name, ((CPPScope*)pyclass)->fCppType))
-                meta_getattro(pyclass, pyname);       // triggers creation
-        }
+    if (!CPyCppyy::CPPDataMember_Check(pyval) && !CPyCppyy::CPPScope_Check(pyval)) {
+        std::string name = CPyCppyy_PyText_AsString(pyname);
+        if (Cppyy::GetNamed(name, ((CPPScope*)pyclass)->fCppType))
+            meta_getattro(pyclass, pyname);       // triggers creation
```

Comment:
This makes sure that `meta_getattro` is called. Later `tp_setattro` calls `dm_set`. Raising the TypeError. Otherwise `tp_setattro` calls generic setattr.